server/scripts/2020TaxSeasonDataScript.php

```diff
@@ -0,0 +1,409 @@
+        $thursdayDates = getWeeklyDatesFromRange('2020-01-30', '2020-04-09');
+        foreach ($dates as $date) {
+			$shiftId = insertShift("$date 17:00:00", "$date 20:00:00", $siteId);
+
+			$firstAppointmentTimeId = insertAppointmentTime("$date 17:00:00", 6, 6, 100, 60, $siteId);
+            $secondAppointmentTimeId = insertAppointmentTime("$date 18:00:00", 6, 6, 100, 60, $siteId);
+            $thirdAppointmentTimeId = insertAppointmentTime("$date 19:00:00", 5, 5, 100, 60, $siteId);
+        }
+
+		// Default Site Role Limits
+		insertSiteRoleLimit(1, $siteCoordinatorRoleId, $siteId);
+		insertSiteRoleLimit(1, $greeterRoleId, $siteId);
+
+		$DB_CONN->commit();
+	} catch (Exception $e) {
+		$DB_CONN->rollback();
+		throw new Exception('Failed inserting Anderson Library data', MY_EXCEPTION);
+		die();
+	}
+}
+
+function insertCenterForPeopleInNeedData() {
+	GLOBAL $DB_CONN;
+
+	$siteCoordinatorRoleId = 1;
+	$greeterRoleId = 2;
+	$preparerRoleId = 3;
+    $reviewerRoleId = 4;
+	
+	$dataAlreadyInserted = true;
+	if ($dataAlreadyInserted) {
+		die('The Center for People in Need data has already been inserted');
+	}
+
+	try {
+		$DB_CONN->beginTransaction();
+
+        $siteId = 5; // Manually obtained from PROD DB
+
+		// Tuesdays
+		$dates = getWeeklyDatesFromRange('2020-02-04', '2020-03-24');
+		foreach ($dates as $date) {
+			$shiftId = insertShift("$date 11:00:00", "$date 14:00:00", $siteId);
+
+			$firstAppointmentTimeId = insertAppointmentTime("$date 11:00:00", 8, 8, 100, 60, $siteId);
+			$secondAppointmentTimeId = insertAppointmentTime("$date 12:00:00", 8, 8, 100, 60, $siteId);
+			$thirdAppointmentTimeId = insertAppointmentTime("$date 13:00:00", 4, 4, 100, 60, $siteId);
+		}
+
+		// Default Site Role Limits
+		insertSiteRoleLimit(1, $siteCoordinatorRoleId, $siteId);
+		insertSiteRoleLimit(1, $greeterRoleId, $siteId);
+
+		$DB_CONN->commit();
+	} catch (Exception $e) {
+		$DB_CONN->rollback();
+		throw new Exception('Failed inserting Center for People in Need data', MY_EXCEPTION);
+		die();
+	}
+}
+
+function insertLorenEiseleyLibraryData() {
+	GLOBAL $DB_CONN;
+
+	$siteCoordinatorRoleId = 1;
+	$greeterRoleId = 2;
+	$preparerRoleId = 3;
+    $reviewerRoleId = 4;
+    
+	$dataAlreadyInserted = true;
+	if ($dataAlreadyInserted) {
+		die('The Eiseley data has already been inserted');
+	}
+
+	try {
+		$DB_CONN->beginTransaction();
+
+        $siteId = 6; // Manually obtained from PROD DB
+
+		// Thursdays
+		$dates = getWeeklyDatesFromRange('2020-02-06', '2020-04-09');
+		foreach ($dates as $date) {
+			$shiftId = insertShift("$date 16:00:00", "$date 19:00:00", $siteId);
+
+			$firstAppointmentTimeId = insertAppointmentTime("$date 16:00:00", 8, $8, 100, 60, $siteId);
+			$secondAppointmentTimeId = insertAppointmentTime("$date 17:00:00", 8, 8, 100, 60, $siteId);
+			$thirdAppointmentTimeId = insertAppointmentTime("$date 18:00:00", 4, 4, 100, 60, $siteId);
+		}
+
+		// Saturdays
+		$dates = getWeeklyDatesFromRange('2020-02-01', '2020-04-11')

```

Comment:
Also, 2020 is a leap year. Does this `getWeeklyDatesFromRange` account for this? i.e. we should have a `2020-02-29` date from that method. We should verify this is the case.

Reply:
I'm also concerned about the last date here. The parameters sheet says the site is open until 4/11, but the flier says 4/4 -- we need to verify this with Katie.

Reply:
This does account for a gap year. Changed to April 4.

**"For the Eiseley site, the site schedule flier says that its last Saturday is Apr 4, but the excel sheet said Apr 11. Which is the correct last Saturday date for the Eiseley site? APR 4 (The site schedule is the most accurate. Thank you for noting the discrepancy.)"**